### PR TITLE
Add event data description to README

### PR DESCRIPTION
**Pull request by fcarstens**
This PR adds a description of the js-reporters event data based on #12.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,26 @@ The criteria for picking these event names included:
...
 
+### Event Data
+
+Based on the discussion in [#12](https://github.com/js-reporters/js-reporters/issues/12#issuecomment-120483356), there are two basic data structures:
+ - **Test**: A test holds basic information on a single test. It has the following set of required attributes
+   - `testName`: name of the test
+   - `suiteName`: name of the suite the test belongs to
+   - `status`: result of the test. Can `passed`, `failed` or `skipped`
+   - `runtime`: execution time in milliseconds
+   - `errors`: array containing all errors. Depending on the framework, this is a single exception or a list of failed assertions.
```
Comment (jzaefferer) [inline README.md]: Add something like this: Will be empty for status other than failed.

File: README.md [hunk 2]
```diff
@@ -51,6 +51,26 @@ The criteria for picking these event names included:
...
+ - **Suite**: A suite is a collection of tests and potentially other suites.
+   - `name`: name of the suite
+   - `childSuites`: array with all direct subsuites
+   - `tests`: array containing all tests that directly belong to the suite (but not to a child suite)
+   - `runtime`: execution time of the whole suite in milliseconds (including child suites)
+   - `status`: summarized status of the suite
+      - `failed`, if at least one test failed
+      - `skipped`, if all tests in the suite are skipped (and there is at least one skipped test)
+      - `passed`, otherwise (the suite may contain skipped tests, but no failed tests)
```
Comment (jzaefferer) [inline README.md]: Let's be explicit. Replace the "otherwise" and specify the exact conditions.

File: README.md [hunk 3]
```diff
@@ -51,6 +51,26 @@ The criteria for picking these event names included:
...
+   - `name`: name of the suite
+   - `childSuites`: array with all direct subsuites
+   - `tests`: array containing all tests that directly belong to the suite (but not to a child suite)
+   - `runtime`: execution time of the whole suite in milliseconds (including child suites)
+   - `status`: summarized status of the suite
+      - `failed`, if at least one test failed
+      - `skipped`, if all tests in the suite are skipped (and there is at least one skipped test)
+      - `passed`, otherwise (the suite may contain skipped tests, but no failed tests) 
+ 
+For `testStart` and `testEnd`, the corresponding test object is passed to the reporter.  The same applies to `suiteStart` and `suiteEnd` where the matching suite object is passed to the reporter. For `runStart` and `runEnd` a "global" suite object is passed to the reporter, which contains all top-level suites as child suites.
```
Comment (jzaefferer) [inline README.md]: This should mention that properties can be undefined during the *start events. Or mention that while specifying each affected property.

File: README.md [hunk 4]
```diff
@@ -51,6 +51,26 @@ The criteria for picking these event names included:
  - It uses names that are valid JavaScript identifiers, which allows using those as keys in JSON and avoids the need to quote keys in regular JS objects or function calls.
  - It doesn't overlap with any known existing events, so introducing these could be done in parallel to the existing API in each framework, optionally deprecating and eventually removing the previous API.
 
+### Event Data
+
+Based on the discussion in [#12](https://github.com/js-reporters/js-reporters/issues/12#issuecomment-120483356), there are two basic data structures:
```
Comment (jzaefferer) [inline README.md]: Add a bit of motivation, something like: These can represent both flat structures like used by QUnit (plus whatever else does that) as well as nested structures like (...). 

Or add this to the description for Suite?

File: README.md [hunk 5]
```diff
@@ -51,6 +51,26 @@ The criteria for picking these event names included:
...
+ - **Test**: A test holds basic information on a single test. It has the following set of required attributes
```
Comment (jzaefferer) [inline README.md]: Extend this a bit to cover BDD vocabulary. Something like: In BDD, this represents a single `it()`.

**Comment by jzaefferer** [comment]
In addition to the above, we should have at least two examples, one with a flat and one with a multiple levels deep nested structure. Could be addressed in a separate PR, please file an issue in that case.

**Comment by fcarstens** [comment]
Included all suggestions.

File: README.md [hunk 6]
```diff
@@ -14,7 +14,11 @@ https://github.com/js-reporters/js-reporters/issues/
 
 ## Background
 
-We on the [QUnit](http://qunitjs.com/) team have been [discussing](https://github.com/jquery/qunit/issues/531) the possibility of working with other JS test frameworks, especially those that can be run client-side (e.g. Mocha, Jasmine, Intern, Buster, etc.), to agree upon a "Common Reporter Interface" so that we could hopefully share Reporter plugins between testing frameworks. This would also benefit high-level consumers of the frameworks such as Karma, BrowserStack, SauceLabs, Testling, etc.
+We on the [QUnit](http://qunitjs.com/) team have been [discussing](https://github.com/jquery/qunit/issues/531) the
+possibility of working with other JS test frameworks, especially those that can be run client-side (e.g. Mocha, Jasmine,
+Intern, Buster, etc.), to agree upon a "Common Reporter Interface" so that we could hopefully share Reporter plugins
+between testing frameworks. This would also benefit high-level consumers of the frameworks such as Karma, BrowserStack,
+SauceLabs, Testling, etc.
```
Comment (jzaefferer) [inline README.md]: Why did you change everything to hard line breaks? We usually don't use those in markdown, makes it much easier to edit prose.

File: README.md [hunk 7]
```diff
@@ -41,16 +47,51 @@ Based on the discussion in [#1](https://github.com/js-reporters/js-reporters/iss
  - `suiteEnd`:  Triggered at the end of each group of tests within a testsuite.
  - `runEnd`:  Indicates the end of a testsuite, triggered just once.
 
-These only define event names, not the data associated with each event. The data still needs to be specified via further discussion in [#1](https://github.com/js-reporters/js-reporters/issues/1).
+These only define event names, not the data associated with each event. The data still needs to be specified via further
+discussion in [#1](https://github.com/js-reporters/js-reporters/issues/1).
```
Comment (jzaefferer) [inline README.md]: This isn't quite right anymore, since we're adding event data below.

File: README.md [hunk 8]
```diff
@@ -41,16 +47,51 @@ Based on the discussion in [#1](https://github.com/js-reporters/js-reporters/iss
...
+   [#1](https://github.com/js-reporters/js-reporters/issues/1#issuecomment-54841874)
+ - It uses names that are valid JavaScript identifiers, which allows using those as keys in JSON and avoids the need to
+   quote keys in regular JS objects or function calls.
+ - It doesn't overlap with any known existing events, so introducing these could be done in parallel to the existing API
+   in each framework, optionally deprecating and eventually removing the previous API.
+
+### Event Data
+
+Based on the discussion in [#12](https://github.com/js-reporters/js-reporters/issues/12#issuecomment-120483356), there
+are two basic data structures: Suites and Tests. A test represents an atomic test/spec/"`it()`". A suite contains tests
```
Comment (jzaefferer) [inline README.md]: I think the `it()` reads better without the quotes.

File: README.md [hunk 9]
```diff
@@ -41,16 +47,51 @@ Based on the discussion in [#1](https://github.com/js-reporters/js-reporters/iss
...
+and optionally other suites. This can represent both flat structures like used by QUnit as well as nested suites like
+used by Jasmine or Mocha. The data structures are defined as follows:
+
+ - **Test**: A test holds basic information on a single test/spec. It has the following set of required attributes
+   - `testName`: name of the test
+   - `suiteName`: name of the suite the test belongs to
+   - `status`: result of the test. Can `passed`, `failed` or `skipped`
```
Comment (jzaefferer) [inline README.md]: This is redundant with the Suite status below. Let's deduplicate that, either with a separate definition that we references in both places or we define status here and reference it from below.

Comment (jzaefferer) [inline README.md]: Actually, suite status is a bit different. Maybe just explain what "skipped" is?

File: README.md [hunk 10]
```diff
@@ -41,16 +47,51 @@ Based on the discussion in [#1](https://github.com/js-reporters/js-reporters/iss
...
+   in each framework, optionally deprecating and eventually removing the previous API.
+
+### Event Data
+
+Based on the discussion in [#12](https://github.com/js-reporters/js-reporters/issues/12#issuecomment-120483356), there
+are two basic data structures: Suites and Tests. A test represents an atomic test/spec/"`it()`". A suite contains tests
+and optionally other suites. This can represent both flat structures like used by QUnit as well as nested suites like
+used by Jasmine or Mocha. The data structures are defined as follows:
+
+ - **Test**: A test holds basic information on a single test/spec. It has the following set of required attributes
```
Comment (jzaefferer) [inline README.md]: Let's add either a dot or colon at the end of that sentence.

**Comment by jzaefferer** [comment]
Looks good to land!

**Comment by jzaefferer** [comment]
Please reference both issues and PRs in commit messages (at least before merging). Elsewhere we use "Fixes #xxx" for issues, "Closes #xxx" for PRs, could use that here as well.
